src/components/Login/Login.jsx

```diff
@@ -21,6 +21,7 @@ export class Login extends React.Component {
 
     componentDidMount() {
         this.props.authenticateUser();
+        this.props.getURI();
```

Comment:
Instead of calling this, can a failed `authenticateUser` fall just return the URI?